lib/repositories/survey_repository.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:survey/models/survey_info.dart';
+import 'package:survey/services/api/survey/survey_api_service.dart';
+import 'package:survey/services/local_storage/local_storage_service.dart';
+import 'package:survey/services/locator/locator_service.dart';
+
+abstract class SurveyRepository {
+  static const listLocalStorageKey = "survey_repository_list";
+
+  Future<bool> get isSurveysCached;
+
+  Future<List<SurveyInfo>> fetchSurveys({bool force});
+}
+
+class SurveyRepositoryImpl implements SurveyRepository {
+  final SurveyApiService _surveyApiService = locator.get();
+  final LocalStorageService _localStorageService = locator.get();
+
+  @override
+  Future<bool> get isSurveysCached async {
+    final items = await _localStorageService
+        .getListObject<SurveyInfo>(SurveyRepository.listLocalStorageKey);
+    return items != null;
+  }
+
+  @override
+  Future<List<SurveyInfo>> fetchSurveys({bool force = false}) async {
+    // Refer items from local storage
+    if (!force) {
+      final items = await _localStorageService
+          .getListObject<SurveyInfo>(SurveyRepository.listLocalStorageKey);
+      if (items != null) {
+        return items;
+      }
+    }
+
+    // Fetch from server
+    final list = await _surveyApiService.list(params: SurveyListParams());
+
+    // Storage it
+    _localStorageService.setListObject(
+      list.items,
+      key: SurveyRepository.listLocalStorageKey,
+    );
+
+    return list.items;
+  }
```

Comment:
I think we missed 1 case here 🤔  the surveys api has pagination (mean load more surveys), we should load next page until reaching to `page_size` in API, right? 😄

Reply:
@manh-tOh I missed this feature. 
Let's skip it for and I will update it in another PR 👍